src/main/java/SellStockUseCase/SellUseCaseInteractor.java

```diff
@@ -0,0 +1,33 @@
+package SellStockUseCase;
+import APIInterface.StockAPIAccess;
+import APIInterface.StockAPIRequest;
+import APIInterface.StockAPIResponse;
+import entities.Portfolio;
+import java.io.IOException;
+
+public class SellUseCaseInteractor {
+
+    /**
+     * This method is used to sell stocks from a portfolio
+     * @param sell The request object containing the portfolio, symbol, and quantity
+     * @return The response object containing the portfolio, symbol, quantity, and price
+     * @throws IOException
+     */
+
+    public SellOutputResponse sellStock(SellInputRequest sell) throws IOException {
+        Portfolio portfolio = sell.getPortfolio();
+        String symbol = sell.getSymbol();
+        int quantity = sell.getQuantity();
+        boolean possible = portfolio.sellStock(symbol, quantity);
```

Comment:
portfolio.sellStock takes a 3rd parameter, a String representing the username.

Reply:
Fixed!